File: workout-pekko/src/main/scala/workout/domain/Workout.scala
package workout.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto._
import java.time.Instant
import java.util.UUID

case class Workout(
    id: UUID,
    name: String,
    description: Option[String],
    workoutType: WorkoutType,
    durationMinutes: Int,
    caloriesBurned: Option[Int],
    difficulty: Difficulty,
    createdAt: Instant,
    updatedAt: Instant
)

object Workout {
  implicit val encoder: Encoder[Workout] = deriveEncoder[Workout]
  implicit val decoder: Decoder[Workout] = deriveDecoder[Workout]
}

File: workout-spring-webflux/src/main/java/com/workout/app/domain/Workout.java
package com.workout.app.domain;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Table;

import java.time.Instant;
import java.util.UUID;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table("workouts")
public class Workout {
    @Id
    private UUID id;
    private String name;
    private String description;
    private String workoutType;
    private int durationMinutes;
    private Integer caloriesBurned;
    private String difficulty;
    private Instant createdAt;
    private Instant updatedAt;
}
